src/Microsoft.DotNet.CodeFormatting/Analyzers/UnwrittenWritableFieldFixer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class UnwrittenWritableFieldThisFixer : CodeFixProvider
+    {
+        private static SyntaxToken s_readOnlyToken =  SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+            var fieldDeclarationNode = root
+                .FindToken(diagnosticSpan.Start)
+                .Parent
+                .FirstAncestorOrSelf<FieldDeclarationSyntax>();
+
+            Debug.Assert(fieldDeclarationNode != null);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    Resources.UnwrittenWritableFieldFixer_Title,
+                    c => AddReadonlyModifier(context.Document, root, fieldDeclarationNode)),
+                diagnostic);
+        }
+
+        private Task<Document> AddReadonlyModifier(Document document, SyntaxNode root, FieldDeclarationSyntax fieldDeclaration)
+        {
+            FieldDeclarationSyntax newFieldDeclaration = fieldDeclaration
+                .WithModifiers(fieldDeclaration.Modifiers.Add(s_readOnlyToken));
```

Comment:
Not so awesome :) See above. I edited my comment in place when I discovered that it fails if you omit the access specifier.